e2e-tests/portal/cypress/e2e/journeys/maker/amend-facility.spec.js

```diff
@@ -0,0 +1,139 @@
+const MOCK_USERS = require('../../../fixtures/users');
+const { contract } = require('../../pages');
+const { MOCK_BOND, MOCK_LOAN } = require('../../../fixtures/mockFacilities');
+const { DEALS, FACILITY } = require('../../../fixtures/constants');
+const relative = require('../../relativeURL');
+
+const { ADMIN, BANK1_MAKER1, BANK1_CHECKER1 } = MOCK_USERS;
+
+context('Amend a facility', () => {
+  describe('Amend a facility', () => {
+    const MOCK_DEAL = {
+      bankInternalRefName: 'dealId',
+      additionalRefName: 'dealName',
+    };
+
+    let dealWithAmendableFacilities;
+    let amendableBond;
+    let amendableLoan;
+
+    let notAcknowledgedDeal;
+    let notAcknowledgedBond;
+    let notAcknowledgedLoan;
+
+    let dealWithUnissuedFacilities;
+    let unissuedBond;
+    let unissuedLoan;
+
+    const facilitiesToCleanUp = [];
+
+    const createMockFacility = (baseMockFacility, isIssued) => ({
+      ...baseMockFacility,
+      facilityStage: isIssued ? FACILITY.FACILITY_STAGE.ISSUED : FACILITY.FACILITY_STAGE.UNISSUED,
+    });
+
+    const createDealWithFacilities = (setDealCallback, setBondCallback, setLoanCallback, status, areFacilitiesIssued) => {
+      cy.insertOneDeal(MOCK_DEAL, BANK1_MAKER1)
+        .then((insertedDeal) => {
+          const dealUpdates = { ...insertedDeal, status };
+
+          cy.updateDeal(insertedDeal._id, dealUpdates, BANK1_MAKER1)
+            .then((updatedDeal) => {
+              setDealCallback(updatedDeal);
+
+              const mockBond = createMockFacility(MOCK_BOND, areFacilitiesIssued);
+              const mockLoan = createMockFacility(MOCK_LOAN, areFacilitiesIssued);
+
+              cy.createFacilities(updatedDeal._id, [mockBond, mockLoan], BANK1_MAKER1)
+                .then((createdFacilities) => {
+                  setBondCallback(createdFacilities.filter((f) => f.type === 'Bond')[0]);
+                  setLoanCallback(createdFacilities.filter((f) => f.type === 'Loan')[0]);
+
+                  facilitiesToCleanUp.push(...createdFacilities);
+                });
+            });
+        });
+    };
+
+    before(() => {
+      cy.deleteDeals(ADMIN);
```

Comment:
This seems to be done in the other portal cypress tests. It means that your local data gets cleaned out when you run these tests locally.

I'm not sure why it's used, and why other tests clear up the facilities after running but not the deals.